src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -203,4 +204,69 @@ public void testParseFilePermissionChange() throws Exception {
             }
         }
     }
+
+    @Test
+    public void testParseLineChangesAddLine() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            Files.write(helloWorld.toPath(), "line 0\n"
+                    .getBytes(Charset.forName("UTF-8")), StandardOpenOption.APPEND);
+            GitUtils.addAllAndCommit(repository, "add original line");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            Files.write(helloWorld.toPath(), "line 1 added\nline 2 added\n"
+                    .getBytes(Charset.forName("UTF-8")), StandardOpenOption.APPEND);
+            GitUtils.addAllAndCommit(repository, "add line 1, 2");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals("There should be 1 change detected", 1, changes.size());
+            final GitChange expected = ImmutableGitChange.builder()
+                    .path("HelloWorld")
+                    .addLines(1, 2)
+                    .build();
+            assertEquals("The change is not as expected", expected, changes.get(0));
+        }
+    }
+
+    @Test
+    public void testParseLineChangesRemoveLine() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            Files.write(helloWorld.toPath(), "line 0\nline 1 to be removed\nline 2 to be removed\n"
+                    .getBytes(Charset.forName("UTF-8")), StandardOpenOption.APPEND);
+            GitUtils.addAllAndCommit(repository, "add original three lines");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            Files.write(helloWorld.toPath(), "line 0\n"
+                    .getBytes(Charset.forName("UTF-8")), StandardOpenOption.TRUNCATE_EXISTING);
+            GitUtils.addAllAndCommit(repository, "remove two lines");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals("There should be 1 change detected", 1, changes.size());
+            final GitChange expected = ImmutableGitChange.builder()
```

Comment:
I assume this should be ok.
We will probably eventually need some way to examine the contents of these lines and their surrounding text for processing.